modules/imgproc/src/imgwarp.cpp

```diff
@@ -2740,9 +3010,9 @@ class WarpPerspectiveInvoker :
                     {
                         x1 = 0;
 
-                        #if CV_TRY_SSE4_1
-                        if (pwarp_impl_sse4)
-                            pwarp_impl_sse4->processNN(M, xy, X0, Y0, W0, bw);
+                        #if CV_SIMD128_64F
+                        if (pwarp_impl_CV_SIMD)
```

Comment:
I think this check is redundant, because the pointer couldn't be empty. I also think that creation of the class is redundant too, because it stores no additional information. IMO it would be better to just call to static functions `processNN` and `process`.